pkg/controller/gkenetworkparamset/gnpcontroller_validations.go

```diff
@@ -217,6 +217,21 @@ func crossValidateNetworkAndGnp(network *networkv1.Network, params *networkv1.GK
 				ErrorMessage: "L3 type network requires secondary range to be specified in params",
 			}
 		}
+
+		// if the "default" params has new pod ranges added from Multi Pod CIDR,
+		// marks the Network not ready. Only apply to new pod ranges to limit the
+		// reconciliation of GNP and Network object
+		if network.Name == networkv1.DefaultPodNetworkName &&
+			params.Name == networkv1.DefaultPodNetworkName &&
+			originalParams.Status.PodCIDRs != nil &&
+			len(originalParams.Status.PodCIDRs.CIDRBlocks) > 0 &&
+			!sameStringSlice(params.Status.PodCIDRs.CIDRBlocks, originalParams.Status.PodCIDRs.CIDRBlocks) {
+			return &gnpNetworkCrossValidation{
+				IsValid:      false,
+				ErrorReason:  networkv1.GNPParamsNotReady,
+				ErrorMessage: "New Pod ranges in default VPC requires network CIDRs update",
+			}
+		}
 	}
```

Comment:
This is a very unintuitive place to update the status to false. Why can't we do this when we first update the params?

Reply:
Done. Move it out to the place when we update the params